src/luascript.cpp

```diff
@@ -13777,66 +13812,927 @@ int LuaScriptInterface::luaPartySetSharedExperience(lua_State* L)
 // Spells
 int LuaScriptInterface::luaSpellCreate(lua_State* L)
 {
-	// Spell(words, name or id)
-	InstantSpell* spell = nullptr;
+	// Spell(words, name or id) to get an existing spell
+	// Spell(type) ex: Spell(SPELL_INSTANT) or Spell(SPELL_RUNE) to create a new spell
+	SpellType_t type = getNumber<SpellType_t>(L, 2);
+	if (isString(L, 2)) {
+		std::string tmp = asLowerCaseString(getString(L, 2));
+		if (tmp == "instant") {
+			type = SPELL_INSTANT;
+		} else if (tmp == "rune") {
+			type = SPELL_RUNE;
+		}
+	}
+
+	if (type) {
+		switch (type) {
+			case SPELL_INSTANT: {
+				InstantSpell* spell = new InstantSpell(getScriptEnv()->getScriptInterface());
+				if (spell) {
+					spell->fromLua = true;
+					pushUserdata<InstantSpell>(L, spell);
+					setMetatable(L, -1, "Spell");
+					spell->spellType = SPELL_INSTANT;
+				} else {
+					lua_pushnil(L);
+				}
+				break;
+			}
+			case SPELL_RUNE: {
+				RuneSpell* spell = new RuneSpell(getScriptEnv()->getScriptInterface());
+				if (spell) {
```

Comment:
This is always true (`new` throws, never returns `nullptr`).

Reply:
fixed in [9e888a9](https://github.com/otland/forgottenserver/pull/2584/commits/9e888a96bb55d97ef102cb18a8c91e78178c8ecf)